### DISCUSSION TITLE
How to use pocketbase javacript SDK for client auth guards?

### DISCUSSION THREAD

**Discussion by silenceofthewans** [pocketbase/pocketbase#1390] [Q&A] [answered]
I'm kinda confused between `pb.authStore.isValid` & `await pb.collection('users').authRefresh()`. From the docs, it seems like I should use `isValid` and it will always check for the freshest state of my user model in the backend but it doesn't and only checks the localStorage token's validity. I can delete my user in the backend and client will not be aware of this because of stateless auth but if I use `authRefresh()` I can redirect because it will atleast error out when the model doesn't exist. So, I'm curious when would someone use only `isValid`

Currently, I'm doing this in my static CSR SvelteKit's `+layout`:

```
> +layout.svelte

onMount(async () => {
    // if (!pb.authStore.isValid) {
    // 	await goto('/login');
    // 	return;
    // }

    try {
        await pb.collection('users').authRefresh();
    } catch (error) {
        pb.authStore.clear();
        await goto('/login');
        return;
    }

});
``` 
Are there any gotchas I should be aware of with this approach other than that it will result in a fetch request on page load.

**Comment by ganigeorgiev** [comment, marked as answer]
`pb.authStore.isValid` loosely checks the current status of your AuthStore state (aka. whether it has nonemtpy token with unexpired exp claim). **It doesn't perform any server-side calls or validations**.

`await pb.collection('users').authRefresh()` refreshes (and validate) the currently stored auth state. It sends a `POST /api/collections/users/auth-refresh` request.

It depends what you are trying to do, but for client-side apps my usual recommendation is to call `pb.collection('users').authRefresh()` on every full page load/refresh if `pb.authStore.isValid` is true (otherwise you will make an unnecessary `authRefresh` request since it will always fail):

```js
> +layout.svelte

onMount(async () => {
    try {
        // validate server-side the auth state
        pb.authStore.isValid && await pb.collection('users').authRefresh();
    } catch (error) {
        pb.authStore.clear();
    }

    if (!pb.authStore.isValid) {
        await goto('/login');
    }
});
```

You can _trust_ `pb.authStore.isValid` after the `authRefresh()` call.

---

I'm not sure how to explain it better. Suggestions for improving the documentation are welcome.

**Reply by silenceofthewans** [reply]
This is better 👍. I think we should eventually have a 'Common Uses' section under sdk's on docs which can have examples like this. Also, not important and probably breaking but whenever I look at `pb.collection('users').authRefresh()` I feel like I'm going to be refreshing auth for all my users lol and would've preferred `pb.authStore.refresh()` or `pb.auth.refresh()`
